torch/utils/data/_utils/collate.py

```diff
@@ -134,7 +134,7 @@ def default_collate(batch):
             # shared memory tensor to avoid an extra copy
             numel = sum(x.numel() for x in batch)
             storage = elem.storage()._new_shared(numel)
-            out = elem.new(storage)
+            out = elem.new(storage).view(-1, *list(elem.size()))
```

Comment:
When `elem` is a scalar tensor, it won't work because `view(-1, 0)` is not valid input. It should be `view(-1)`.

Reply:
Actually it works. When `elem` is a scalar tensor, `list(elem.size())` returns an empty list `[]` and star operation on an empty list gives null return, thus it turns out to be `view(-1,)` as expected, not `view(-1,0)`. `view(-1,)` works as the same as `view(-1)`.

Test code:
```
import torch

# For case `elem` is an array
#elem_size = [5]
#a, b = torch.randn(elem_size), torch.randn(elem_size)

# For case `elem` is a scalar
a, b = torch.tensor(0.), torch.tensor(0.)

batch = [a, b]
elem = batch[0]

numel = sum(x.numel() for x in batch)
storage = elem.storage()._new_shared(numel)
out = elem.new(storage).view(-1, *list(elem.size()))

torch.stack(batch, 0, out=out)
```

Reply:
Sorry. I mean empty Tensor `torch.randn(0)`

Reply:
Should work now.